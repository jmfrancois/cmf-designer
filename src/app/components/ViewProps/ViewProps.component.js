import React from 'react';
import PropTypes from 'prop-types';
import { cmfConnect } from '@talend/react-cmf';

function renderValue(value) {
	if (Array.isArray(value)) {
		if (value.length === 0) {
			return '[]';
		}
		return (
			<ul>
				{value.map((v, i) => <li key={i}>{i}: {renderValue(v)}</li>)}
			</ul>
		);
	} if (typeof value === 'object') {
		return (
			<ul>
				{Object.keys(value).map((v, i) => <li key={i}>{v}: {renderValue(value[v])}</li>)}
			</ul>
		);
	} if (typeof value === 'string') {
		return value;
	} if (typeof value === 'boolean') {
		return value ? 'true' : 'false';
	}
	return value;
}


function ViewProps(props) {
	if (!props.item) {
		return null;
	}
	const item = props.item.toJS();
	if (item.value === undefined) {
		return null;
	}
	const metadata = item.name.split('#');
	return (
		<div>
			{/* <button
				className="btn btn-danger pull-right"
				onClick={() => props.dispatch({
					type: ViewProps.ACTION_TYPE_DELETE,
					id: item.get('id'),
				})}
			>
				Delete
			</button> */}
			<h1>{item.name}</h1>
			{metadata.length !== 2 && (
				<div className="alert alert-warning">
					<p>You should use the following syntax in the props id:</p>
					<pre>ComponentName#componentId</pre>
				</div>
			)}
			<ul>
				{Object.keys(item.value).map((key, index) => (
					<li key={index}>{key}: {renderValue(item.value[key])}</li>
				))}
			</ul>
		</div>
	);
}

ViewProps.propTypes = {
	name: PropTypes.string,
	...cmfConnect.propTypes,
};
ViewProps.displayName = 'ViewProps';

export default cmfConnect({})(ViewProps);
