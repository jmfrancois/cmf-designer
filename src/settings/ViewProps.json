{
  "props": {
    "ViewProps#default": {
      "itemExpression": "getActiveProps"
    }
  }
}
