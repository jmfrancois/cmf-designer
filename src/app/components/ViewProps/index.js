import ViewProps from './ViewProps.component';

export default ViewProps;
