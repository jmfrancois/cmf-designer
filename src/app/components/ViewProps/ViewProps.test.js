import React from 'react';
import { shallow } from 'enzyme';

import ViewProps from './ViewProps.component';

describe('ViewProps', () => {
	it('should render', () => {
		const wrapper = shallow(
			<ViewProps />
		);
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
