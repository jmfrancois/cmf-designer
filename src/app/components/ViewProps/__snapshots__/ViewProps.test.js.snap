// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewProps should render 1`] = `null`;
